titan-runtime/titan.c

```diff
@@ -204,3 +205,13 @@ const TValue *getgeneric (Table *t, const TValue *key) {
     }
   }
 }
+
+int errorhandler(lua_State *L) {
+  void *buffer[255];
+  const int calls = backtrace(buffer, sizeof(buffer) / sizeof(void *));
```

Comment:
We need some #ifdef around this to maintain portability. Even on Linux there are systems without `execinfo.h` (any non-glibc system, which includes most of embedded Linux and Alpine for small containers).

htop uses #ifdef HAVE_EXECINFO_H (which its `configure` script sets). Even if we don't put the actual detection for now, it's good to mark the knowingly-unportable code from the beginning so we don't have to hunt for it later. :)

Reply:
Oh, I am not intending to merge this as is, this was just a quick experiment, the next step as I said is to write the code to build the stack trace object, and not to print it. I just wanted to see if I could attach an error handler before the stack unwinds and if it seemed to have all necessary information available.

Reply:
Let's merge as is! Small PRs mean easy reviews :)

Reply:
Even if we don't do the full integrated stack trace having the two stacktraces is already a useful thing.

Reply:
I have amended the PR to not print the stack trace, but save it in the registry. There is a stub `titan.so` module with a single `backtrace()` function that reads this value. The backtrace that is saved is an amalgamation of the error message + Lua backtrace + C backtrace. Also added a test case.